app/src/main/java/com/lagradost/cloudstream3/ui/subtitles/SubtitlesFragment.kt

```diff
@@ -415,6 +417,23 @@ class SubtitlesFragment : Fragment() {
             }
         }
 
+        subtitles_disable_subs?.isChecked = state.removeSubs
+        subtitles_disable_subs?.setOnCheckedChangeListener { _, b ->
+            state.removeSubs = b
+            if (b) {
+                setKey(
+                    PRIOR_SUB_LANGUAGE,
+                    getKey(SUBTITLE_AUTO_SELECT_KEY) ?: "en"
+                )
+                setKey(SUBTITLE_AUTO_SELECT_KEY, "")
+            } else {
+                setKey(
+                    SUBTITLE_AUTO_SELECT_KEY,
+                    getKey(PRIOR_SUB_LANGUAGE) ?: "en"
```

Comment:
This can cause unintended behavior if I am understanding this correctly.

1. Lets say you have the following setup:

Preferred language: English
Disable subs: OFF

2. You toggle the switch

Saved Language: English
Preferred language: None ("")
Disable subs: ON

3. You change subtitle language

Saved Language: English
Preferred language: Arabic
Disable subs: ON

4. You enable subs and your preference will get removed:

Saved Language: English
Preferred language: English
Disable subs: OFF

This is why clarification is needed, since even such simple code can easily have hard to spot bugs :pensive:

Reply:
Huh. Here I was assuming the person would want the language saved before-hand. I guess the the toggle should automatically be changed back to off if a language is selected? Or is there some reason why someone would change the language while still wanting subs to be off?

Reply:
Yes, I think changing to off if no language is selected is good, but make sure that the UI updates when the language is changed.